### ISSUE TITLE
Slingshot Dependency

### ISSUE THREAD

**Issue by rwilson** [davidsantiago/stencil#36]
__Suggestion:__ consider removing the slingshot dependency. Two reasons: it's on an old version, (0.10.3 vs the current 0.12.2) and the usage is fairly limited. Removing it would be 1 fewer dependency to conflict with other libs that also depend on slingshot, without sacrificing anything.

#### Details
It looks like stencil only uses slingshot to:
  1. Use`throw+` to throw a string, the `stencil.loader/no-core-cache-msg`
  2. Throw an actual exception in `stencil.parser/get-line-col-from-index`
  3. Throw maps in `stencil.parser/parse` and `stencil.parser/parse-tag`

But, I don't see a case of stencil using slingshot's catch destructuring, instead only catching actual throwables while the maps and strings thrown leak out to the caller. Replacing `no-core-cache-msg` with a plain old `Exception` and the maps with `ex-info` would be a minimal functionality difference but eliminate the dependency.

**Comment by davidsantiago** [comment]
Good point. Would accept a patch for this.

**Comment by rwilson** [comment]
Done, it's #37 if you want it. I didn't foresee that it would require dropping support for clojure < 1.4, so it's understandable if you'd prefer to keep support for older clojure versions instead of reducing dependencies.

**Comment by davidsantiago** [comment]
Thanks for the really well-done and complete patch. I released 0.5.0 with your changes.

I'm happy to get rid of the slingshot dependency. Nothing against slingshot itself, but I added that at a time when I had been convinced that slingshot was going to be the Clojure way to return more than a string in an exception. It didn't work out that way, so it's hard to justify a library just to provide something Clojure has now provided for 3 years.
